src/pages/ServiceDetails/ServiceInfo.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import ServiceId from '../../types/ServiceId';
+import { ServiceInfoPods, ServiceInfoRules, ServiceInfoRoutes, ServiceInfoDescription } from './ServiceInfo/';
+import { Endpoints, Pod, Port, Source, Rule } from '../../types/ServiceInfo';
+import * as API from '../../services/Api';
+import { ToastNotification, ToastNotificationList, Col, Row } from 'patternfly-react';
+
+type ServiceInfoState = {
+  labels?: Map<string, string>;
+  type: string;
+  name: string;
+  ip: string;
+  ports?: Port[];
+  endpoints?: Endpoints[];
+  pods?: Pod[];
+  rules?: Rule[];
+  dependencies?: Source[];
+  error: boolean;
+  errorMessage: string;
+};
+
+class ServiceInfo extends React.Component<ServiceId, ServiceInfoState> {
+  constructor(props: ServiceId) {
+    super(props);
+    this.state = {
+      labels: new Map(),
+      name: '',
+      type: '',
+      ip: '',
+      ports: [],
+      pods: [],
+      rules: [],
+      dependencies: [],
+      error: false,
+      errorMessage: ''
+    };
+  }
+
+  componentWillMount() {
+    this.fetchServiceDetails(this.props);
+  }
+
+  componentWillReceiveProps(nextProps: ServiceId) {
+    this.fetchServiceDetails(nextProps);
+  }
+
+  fetchServiceDetails(props: ServiceId) {
+    console.log('Fetching info of a service...');
+    API.GetServiceDetail(props.namespace, props.service)
+      .then(response => {
+        console.log(response['data']);
+        let data = response['data'];
+        this.setState({
+          labels: data.labels,
+          name: data.name,
+          type: data.type,
+          ports: data.ports,
+          endpoints: data.endpoints,
+          pods: data.pods,
+          dependencies: data.dependencies,
+          rules: data.route_rules,
+          ip: data.ip
+        });
+        console.log(this.state.rules);
+      })
+      .catch(error => {
+        this.setState({
+          error: true,
+          errorMessage: 'Could not connect to server'
+        });
+        console.log(error);
+      });
+  }
+
+  render() {
+    return (
+      <div>
+        {this.state.error ? (
+          <ToastNotificationList>
```

Comment:
Minor suggestion,  not a blocker, just let me know what do you think.
I guess ToastNotificationList can be extracted in a component that can be reused, in the service list page as well, and perhaps in the graph.
As this PR is a main refactor, I guess it could make sense to apply here (and then we unify them in all pages).